quarkus-workshop-super-heroes/docs/src/docs/asciidoc/5a-contract-testing/happy-path-consumer-tests.adoc

```diff
@@ -0,0 +1,166 @@
+
+
+== The first contract test
+
+icon:hand-point-right[role="red",size=2x] [red big]#Call to action#
+
+Add the Quarkus extension to the `pom.xml` of the Fights service:
+
+[source,xml]
+----
+        <dependency>
+            <groupId>io.quarkiverse.pact</groupId>
+            <artifactId>quarkus-pact-consumer</artifactId>
+            <version>{quarkus-pact-version}</version>
+            <scope>test</scope>
+        </dependency>
+----
+
+
+The Pact framework will stand up a server which listens on a port.
+To avoid port conflicts between the real services and the Pact stubs, let's use a different port for the contract instances.
+Open the fight service's `application.properties` and add the following
+
+----
+%test.io.quarkus.workshop.superheroes.fight.client.HeroProxy/mp-rest/url=http://localhost:8093
+----
+
+The `%test` scopes the updated property to only apply in the test mode.
+
+We will be using the https://docs.pact.io/implementation_guides/jvm/consumer/junit5[junit 5 pact libraries].
+
+Create a class called `FightResourceConsumerTest.java`.
+
+[source,java]
+----
+package io.quarkus.workshop.superheroes.fight;
+
+import au.com.dius.pact.consumer.dsl.PactDslWithProvider;
+import au.com.dius.pact.consumer.junit5.PactConsumerTestExt;
+import au.com.dius.pact.consumer.junit5.PactTestFor;
+import au.com.dius.pact.core.model.PactSpecVersion;
+import au.com.dius.pact.core.model.V4Pact;
+import au.com.dius.pact.core.model.annotations.Pact;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.workshop.superheroes.fight.client.DefaultTestVillain;
+import org.hamcrest.core.Is;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import java.util.Map;
+
+import static au.com.dius.pact.consumer.dsl.LambdaDsl.newJsonBody;
+import static io.quarkus.workshop.superheroes.fight.client.DefaultTestHero.DEFAULT_HERO_LEVEL;
+import static io.quarkus.workshop.superheroes.fight.client.DefaultTestHero.DEFAULT_HERO_NAME;
+import static io.quarkus.workshop.superheroes.fight.client.DefaultTestHero.DEFAULT_HERO_PICTURE;
+import static io.restassured.RestAssured.given;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.Response.Status.OK;
+
+@QuarkusTest
+@ExtendWith(PactConsumerTestExt.class)
+@PactTestFor(
+    providerName = "rest-heroes",
+    // Hard-code the Pact MockServer to what we defined in application.properties
+    port = "8093"
+)
+public class FightResourceConsumerTest {
+
+    protected static final String HERO_API_BASE_URI = "/api/heroes";
```

Comment:
Why `protected` and not `private`?

Reply:
No good reason, fixed!

Reply:
(I think I must have had an idea of inheritance between contract and conventional tests at some point, but then I abandoned it and just made the data-carrying object instead.)